lib/palindrome_permutation.rb

```diff
@@ -1,4 +1,23 @@
 
 def palindrome_permutation?(string)
+  hash = {}
+  leng = string.length
+
+  i = 0
+  while i < leng
+    if hash[string[i]]
+      hash[string[i]] += 1
+    else
+      hash[string[i]] = 1
```

Comment:
Since `string[i]` is referenced three times here it would be good to extract it to a variable before the `if` statement. A good name might be `input_char`.

Reply:
This `while` loop definitely gets the job done. A little more Ruby-ish way to loop through the string's characters would be

```
string.chars.each do |char|
  if hash[char]
    hash[char] += 1
  else
    hash[char] = 1
  end
end
```